packages/devtools_app/lib/src/shared/memory/class_name.dart

```diff
@@ -130,14 +130,22 @@ class HeapClassName {
     // class names.
     assert(false, 'Unexpected library for $className: $library.');
     return false;
-  }
+  }();
 
-  String get shortName {
+  late final shortName = () {
     if (className == 'Context' && library == '') return 'Closure Context';
     return className;
-  }
+  }();
+
+  ClassType? _cachedClassType;
 
   ClassType classType(String? rootPackage) {
+    if (_cachedClassType != null) return _cachedClassType!;
+    _cachedClassType = _classType(rootPackage);
+    return _cachedClassType!;
```

Comment:
This can be `return _cachedClassType ??= _classType(rootPackage)!`

Reply:
resolved in https://github.com/flutter/devtools/pull/5134